gpytorch/lazy/lazy_variable.py

```diff
@@ -400,20 +400,26 @@ def exact_predictive_mean(self, full_mean, train_labels, noise, precomputed_cach
         Returns:
             :obj:`torch.tensor`: The predictive posterior mean of the test points
         """
-        n_train = train_labels.size(-1)
+        from ..random_variables import GaussianRandomVariable
+
         if precomputed_cache is None:
             train_mean = full_mean.narrow(-1, 0, n_train)
             if self.ndimension() == 3:
-                train_train_covar = self[:, :n_train, :n_train].add_diag(noise)
+                train_train_covar = self[:, :n_train, :n_train]
```

Comment:
enough to assign train_train_covar in the if/else clauses, no need to repeat the rest of the code